legos/xkcd.py

```diff
@@ -1,15 +1,20 @@
 from Legobot.Lego import Lego
 import requests
-import re
 import logging
+import json
+import random
 
 logger = logging.getLogger(__name__)
 
 
 class XKCD(Lego):
     def listening_for(self, message):
-        return message['text'].split()[0] == '!xkcd'
-        logger.debug('xkcd lego triggered')
+        if message['text'] is not None:
+            try:
+                return message['text'].split()[0] == '!xkcd'
+            except Exception as e:
+                logger.error('XKCD lego failed to check message text: %s' % e)
```

Comment:
While we're already changing a bunch of stuff, let's convert this to the format() style interpolation